test/fuzz/gen.ml

```diff
@@ -139,28 +139,37 @@ let expr_available_2_f64 =
 
 let if_else expr ~locals ~stack env =
   (* TODO: finish > bug typechecking + List.rev *)
+
+  (* se comporte mieux grace au "rev" dans expr, mais pas encore ça!
+      pb avec le fuel à priori : tout passe dans le 1er
+      pt stack
+      pas sûr que pt soit bien utilisé ? jouer avec let * + ?   
+  *)
+
   match stack with
   | Num_type I32 :: _stack -> begin
-    let rt = [] in
-    (* let* rt = list B.val_type in *)
+    (* let rt = [] in *)
+    let* rt = list B.val_type in
     let pt = [] in
-    (* let pt = List.rev stack in *)
+    (* let pt = stack in *)
     (* TODO: take only a prefix *)
-    let typ1 = Arg.Bt_raw (None, (List.rev_map (fun t -> (None, t)) pt, rt)) in
-    let typ2 = Arg.Bt_raw (None, (List.rev_map (fun t -> (None, t)) pt, rt)) in
-    (* let typ1 = Arg.Bt_raw (None, (List.map (fun t -> (None, t)) pt, rt)) in
-       let _typ2 = Arg.Bt_raw (None, (List.map (fun t -> (None, t)) pt, rt)) in *)
+    (* let typ1 = Arg.Bt_raw (None, (List.rev_map (fun t -> (None, t)) pt, rt)) in
+    let typ2 = Arg.Bt_raw (None, (List.rev_map (fun t -> (None, t)) pt, rt)) in *)
+    let typ1 = Arg.Bt_raw (None, (List.map (fun t -> (None, t)) pt, rt)) in
+    let typ2 = Arg.Bt_raw (None, (List.map (fun t -> (None, t)) pt, rt)) in
     (* let pt = List.rev pt in *)
-    let* expr_then = expr ~block_type:typ1 ~stack:pt ~locals env in
-    let* expr_else = expr ~block_type:typ2 ~stack:pt ~locals env in
+    
+    let* expr_then = expr ~block_type:typ1 ~stack:pt ~locals ~start:false env in
+    let* expr_else = expr ~block_type:typ2 ~stack:pt ~locals ~start:false env in
+    
     let instr = If_else (None, Some typ1, expr_then, expr_else) in
     let pt_descr = S.Pop :: List.map (fun _ -> S.Pop) pt in
     let rt_descr = List.map (fun t -> S.Push t) rt in
     pair (const instr) (const (pt_descr @ rt_descr))
   end
   | _ -> assert false
 
-let rec expr ~block_type ~stack ~locals env =
+let rec expr ~block_type ~stack ~locals ~start env =
```

Comment:
I'm not convinced by this `start` thing. This is to avoid infinite loops, right ?

Reply:
`start` is a boolean parameter just to say: consider function calls only in the $start function
I don't know if it's the right way to do it ?

Reply:
Why would we only call functions in the `$start` function ?

Reply:
An initial restriction, to avoid recurcive calls

Reply:
`start` parameter has been removed.
Now, function calls (but not recursive) can be made anywhere.